### ISSUE TITLE
[Mastodon] OAuth flow doesn't complete in Raycast

### ISSUE THREAD

**Issue by th0rgall** [raycast/extensions#8513]
### Extension

https://www.raycast.com/SevicheCC/mastodon

### Description

I recently noticed the Mastodon extension in the store (amazing!), but I'm not able to log in on my Mastodon server using it.

I'm not sure if this is an extension bug. I forked the extension and tried the below reproduction also on the extension dev server, but I didn't get any error logs there. So maybe it's related to the OAuth client API of Raycast itself? ~~Or something else?~~

### Steps To Reproduce

See the reproduction video: https://github.com/raycast/extensions/assets/11543641/baba13b9-06e7-44cc-89fb-cc1bf5032cdb

1. Set a server URL for the extension (in my case: indieweb.social)
2. Try to run an extension command (Add Simple Status)
3. Click "Sign in with Mastodon" (redirects to my server oauth screen as expected)
4. Get redirected to the Raycast OAuth success screen (as expected)
6. Raycast opens/is activated, but the extension doesn't seem to have received auth credentials.

### Expected Behaviour

The OAuth flow should complete within Raycast, and the extension should be logged in.

### _Update: workaround_

I worked around this issue by invoking a different extension command than "Add Simple Status". See the comments below for additional debugging info.

**Comment by Sevichecc** [comment]
Sorry, I have recently registered an account on `indieweb.social` and attempted to use it with the extension. But I was unable to reproduce the issue, as everything appears to be functioning properly. You may want to check your internet connection or try uninstalling and reinstalling the extension.

**Comment by th0rgall** [comment]
Thank you for trying to reproduce it! I'm sure my internet is working (as I'm able to post this comment). I tried uninstalling & reinstalling now, to no avail.

But as I mentioned:
> I'm not sure if this is an extension bug. I forked the extension and tried the below reproduction also on the extension dev server, but I didn't get any error logs there.

I do remember though that the Vercel extension OAuth flow completed without problems since, so that discounts the "general Raycast OAuth client API is broken on my machine" theory. Weird!

I'll try to dig some more in my fork or Raycast debugging, to see if I can get more information.

**Comment by th0rgall** [comment]
I'm adding a Raycast log of the issue, retrieved with this method https://manual.raycast.com/troubleshooting/getting-logs-via-consoleapp

[ray.20231013_103827 copy.log](https://github.com/raycast/extensions/files/12890845/ray.20231013_103827.copy.log)

There is one interesting line:

> 2023-10-13 10:38:47.504 Df Raycast[663:1481] [com.raycast.macos:app] **Unhandled Raycast scheme call**
>		URL: raycast://oauth?code=[redacted]-[redacted-[redacted]&package_name=Extension&state=[redacted]

I redacted keys/identifiers from the log. The scheme call isn't handled. But which application part is supposed to handle it? Raycast or the extension?

Is `&package_name=Extension` also normal? That seems too generic for it to be able to open the Mastodon extension 🤔

**Comment by th0rgall** [comment]
Update:I got it to work by initiating the OAuth flow through the command "View My Status" rather than "Add Simple Status".

Before trying that, I tried debugging my local fork further:
-  I noticed that this promise didn't resolve (hence: a Raycast issue, not an extension issue) https://github.com/raycast/extensions/blob/1824339d4b3b404efb53e4fb09ac79d190437773/extensions/mastodon/src/utils/oauth.ts#L72 -> this points at a Raycast API issue (just like the log above?)
- then I believe I tried "Add Status", which worked, leading me to trying "View My Status" in the main extension.

Not sure how this difference in commands could have caused the issue.
